service/history/api/recordchildworkflowcompleted/api.go

```diff
@@ -93,11 +93,29 @@ func Invoke(
 
 			// Check mutable state to make sure child execution is in pending child executions
 			ci, isRunning := mutableState.GetChildExecutionInfo(parentInitiatedID)
-			if !isRunning || ci.StartedEventId == common.EmptyEventID {
-				// note we already checked if startedEventID is empty (in consistency predicate)
-				// and reloaded mutable state
+			if !isRunning {
```

Comment:
I think the StartEventID check can also be removed from L76?

Reply:
My thinking was we would still want to fail the consistency predicate and reload the newest mutable state to avoid potential conflicts

Reply:
To avoid potential conflict, we are paying the price of an additional read. Comparing the cost of those two, I would actually choose the first one. The possibility of stale mutable state and child record completed before started are both low today. But soon, the later will happen more often. So we might do more read when there's no conflict.

From different perspectives: 
- The main purpose of the consistency check is to prevent the API handler logic becomes a read only one, otherwise we may never know if the state is stale or not. Since we will also add the started event and perform a write operation, that case won't happen (if initiated event is there.) 
- I actually view consistency check as a causality check, and the record completion call no longer casually depends on the started event

Reply:
Makes sense. Thanks for the explanation.